src/demo-app/dialog/dialog-demo.ts

```diff
@@ -1,7 +1,21 @@
 import {Component, Inject, ViewChild, TemplateRef} from '@angular/core';
 import {DOCUMENT} from '@angular/platform-browser';
-import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from '@angular/material';
-
+import {
+  DialogPosition,
+  MatDialog,
+  MatDialogConfig,
+  MatDialogRef,
+  MAT_DIALOG_DATA
+} from '@angular/material';
+
+/**
+ * Appeases the AOT type checker for the template by extending `MatDialogConfig`
```

Comment:
I would rather not add a separate type just for the sake of AoT. You can remove the type from the `config`. It'll still be type-checked when it's passed in to `dialog.open`.

Reply:
Correct, `dialog.open` would type-check, but not having type results in AOT errors because the template would only have the inferred type to go off from:
`Error at /home/travis/build/angular/material2/dist/packages/demo-app/dialog/dialog-demo.ngfactory.ts:526:51: Property 'minWidth' does not exist on type '{ disableClose: boolean; panelClass: string; hasBackdrop: boolean; backdropClass: string; width: ...'.`
See https://travis-ci.org/angular/material2/jobs/282614008.

Reply:
You can get it to play nice if you initialize the `minWidth` to something.

Reply:
@crisbeto another alternative would be to just type with `MatDialogConfig` without extending, but given strictNullChecks, AOT will complain in the template about `config.property` possibly being `undefined`, so something like the below could be done. Let me know your thoughts.
```html
<ng-template [ngIf]="config.position">
  <p>
    <mat-form-field>
      <input matInput [(ngModel)]="config.position.top" (change)="config.position.bottom = ''" placeholder="Top">
    </mat-form-field>
    <mat-form-field>
      <input matInput [(ngModel)]="config.position.bottom" (change)="config.position.top = ''" placeholder="Bottom">
    </mat-form-field>
  </p>

  <p>
    <mat-form-field>
      <input matInput [(ngModel)]="config.position.left" (change)="config.position.right = ''" placeholder="Left">
    </mat-form-field>
    <mat-form-field>
      <input matInput [(ngModel)]="config.position.right" (change)="config.position.left = ''" placeholder="Right">
    </mat-form-field>
  </p>
</ng-template>
```

Reply:
I think that's over-complicating it. Initializing the new properties to empty strings should be fine.

Reply:
Agreed re: initializing the `config` props. The only question I'd have is around setting the value for `maxWidth` since I'm initializing with `80vw` on DialogConfig. Initializing it to something neutral like an empty string would override the value.

To your point about initializing, it could be as simple as initializing all min/max values with an empty string and initializing `maxWidth` with the `80vw`, or extracting the default value out of a new `DialogConfig` and setting that in the `maxWidth` init.

Reply:
Initializing all of them to an empty string (only in the demo) and the `maxWidth` to the same as the default is fine by me.

Reply:
Cool. Addressed in 2b1c7334157c16e0195557de2f537f24d3e27f6f